src/wallet/wallet.cpp

```diff
@@ -2028,9 +2072,8 @@ bool CWallet::SelectorMatchesAddress(
         [&](const libzcash::UnifiedFullViewingKey& ufvk) {
             auto saplingKey = ufvk.GetSaplingKey();
             if (saplingKey.has_value()) {
-                auto j = saplingKey.value().DecryptDiversifier(a0.d);
-                auto addr = saplingKey.value().Address(j);
-                return addr.has_value() && addr.value() == a0;
+                auto j = saplingKey.value().DecryptDiversifier(a0);
```

Comment:
Ditto here.

Reply:
Same - this authenticates by attempting to reproduce the original address.